Improve the Migration Guide (#5391)
containous/traefik: docs/content/migration/v1-to-v2.md

      - "traefik.http.routers.router0.tls.options=myTLSOptions@file"
    ```

> ## HTTP to HTTPS Redirection is now applied on Router

Reviewer: ```suggestion
## HTTP to HTTPS Redirection is now configured on Routers
```

        
    [http.services]
      [[http.services.my-service.loadBalancer.servers]]
        url = "http://10.10.10.1:80"
      [[http.services.my-service.loadBalancer.servers]]
        url = "http://10.10.10.2:80"
    
    [http.middlewares]
      [http.middlewares.redirect.redirectScheme]
        scheme = "https"
    
    [[tls.certificates]]
          certFile = "/path/to/domain.cert"
          keyFile = "/path/to/domain.key"    
    ```
    
    ```yaml tab="File (YAML)"
    http:
      routers:
        router0:
            entryPoints:
            - web
            middlewares:
            - redirect
            service: my-service
    
        router1:
            entryPoints:
                - web-secure
            service: my-service
            tls: {}
    
      services:
        my-service:
          loadBalancer:
            servers:
            - url: http://10.10.10.1:80
            - url: http://10.10.10.2:80
    
      middlewares:
        redirect:
          redirectScheme:
            scheme: https
    
    tls:
      certificate:
      - certFile: /app/certs/server/server.pem
        keyFile: /app/certs/server/server.pem
    ``` 

> ## ACME (let's encrypt)

Reviewer: ```suggestion
## ACME (LetsEncrypt)
```

    --certificatesResolvers.sample.acme.storage: acme.json
    --certificatesResolvers.sample.acme.httpChallenge.entryPoint: web
    ```

## Traefik Logs

In the v2, all the [log configuration](../observability/logs.md) remains in the static part but are unified under a `log` section.
There is no more log configuration at the root level.

!!! example "Simple log configuration"

    ### v1
    
    ```toml tab="File (TOML)"
    # static configuration
    logLevel = "DEBUG"
    
    [traefikLog]
      filePath = "/path/to/traefik.log"
      format   = "json"
    ```
    
    ```bash tab="CLI"
    --logLevel="DEBUG"
    --traefikLog.filePath="/path/to/traefik.log"
    --traefikLog.format="json"
    ```
    
    ### v2
    
    ```toml tab="File (TOML)"
    # static configuration
    [log]
      level = "DEBUG"
      filePath = "/path/to/log-file.log"
      format = "json"
    ```
    
    ```yaml tab="File (YAML)"
    # static configuration
    log:
      level: DEBUG
      filePath: /path/to/log-file.log
      format: json
    ``` 
    
    ```bash tab="CLI"
    --log.level="DEBUG"
    --log.filePath="/path/to/traefik.log"
    --log.format="json"
    ```

## Tracing

> Traefik v2 remains the OpenTracing support. The `backend` root option from the v1 is gone, you just have to set your [tracing configuration](../observability/tracing/overview.md).

Reviewer: ```suggestion
Traefik v2 retains OpenTracing support. The `backend` root option from the v1 is gone, you just have to set your [tracing configuration](../observability/tracing/overview.md).
```

    --tracing.backend="jaeger"
    --tracing.servicename="tracing"
    --tracing.jaeger.localagenthostport="12.0.0.1:6831"
    --tracing.jaeger.samplingparam="1.0"
    --tracing.jaeger.samplingserverurl="http://12.0.0.1:5778/sampling"
    --tracing.jaeger.samplingtype="const" 
    ```
    
    ### v2
    
    ```toml tab="File (TOML)"
    # static configuration
    [tracing]
      servicename = "tracing"
      [tracing.jaeger]
        samplingParam = 1.0
        samplingServerURL = "http://12.0.0.1:5778/sampling"
        samplingType = "const"
        localAgentHostPort = "12.0.0.1:6831"
     ```
    
    ```yaml tab="File (YAML)"
    # static configuration
    tracing:
      servicename: tracing
      jaeger:
        samplingParam: 1
        samplingServerURL: 'http://12.0.0.1:5778/sampling'
        samplingType: const
        localAgentHostPort: '12.0.0.1:6831'
    ``` 
    
    ```bash tab="CLI"
    --tracing.servicename="tracing"
    --tracing.jaeger.localagenthostport="12.0.0.1:6831"
    --tracing.jaeger.samplingparam="1.0"
    --tracing.jaeger.samplingserverurl="http://12.0.0.1:5778/sampling"
    --tracing.jaeger.samplingtype="const"
    ```

## Metrics

> The v2 remains the metrics tools and allows to configure the metric for the entrypoints and/or the services.

Reviewer: ```suggestion
The v2 retains metrics tools and allows metrics to be configured for the entrypoints and/or services.
```

For a basic configuration, the [metrics configuration](../observability/metrics/overview.md) remains the same.     

!!! example "Simple Prometheus metrics configuration"

    ### v1
    
    ```toml tab="File (TOML)"
    # static configuration
    [metrics.prometheus]
        buckets = [0.1,0.3,1.2,5.0]
        entryPoint = "traefik"
    ```
    
    ```bash tab="CLI"
    --metrics.prometheus.buckets=[0.1,0.3,1.2,5.0]
    --metrics.prometheus.entrypoint="traefik"
    ```
    
    ### v2
    
    ```toml tab="File (TOML)"
    # static configuration
    [metrics.prometheus]
        buckets = [0.1,0.3,1.2,5.0]
        entryPoint = "metrics"
    ```
    
    ```yaml tab="File (YAML)"
    # static configuration
    metrics:
      prometheus:
        buckets:
          - 0.1
          - 0.3
          - 1.2
          - 5
        entryPoint: metrics
    ``` 
    
    ```bash tab="CLI"
    --metrics.prometheus.buckets=[0.1,0.3,1.2,5.0]
    --metrics.prometheus.entrypoint="metrics"
    ```

## No more root level key/values

> To avoid any source of confusion, there is no more configuration at the root level.

Reviewer: ```suggestion
To avoid any source of confusion, there are no more configuration at the root level.
```
